src/libstore/lock.cc

```diff
@@ -123,8 +123,12 @@ struct AutoUserLock : UserLock
 
     std::vector<gid_t> getSupplementaryGIDs() override { return {}; }
 
-    static std::unique_ptr<UserLock> acquire(uid_t nrIds, bool useChroot)
+    static std::unique_ptr<UserLock> acquire(uid_t nrIds, bool useUserNamespace)
     {
+        #if !defined(__linux__)
+        useUserNamespace = false;
+        #endif
```

Comment:
Isn't it a bit weird to conditionally hardcode the value of the parameter? Shouldn't we instead move that logic at the (unique) call site?

Reply:
No, because then the call site can screw it up. That's fine if there is only one call site, but you could imagine multiple users of this class.

Reply:
Couldn't we make it an assertion instead then? It's not too big a deal, but that looks quite confusing